### PR TITLE
[Compression] Add quantization tutorial

### PR DESCRIPTION
**Pull request by Bonytu**
#### Test Options ####
  - [ ] fast test
  - [ ] full test - HPO
  - [ ] full test - NAS
  - [ ] full test - compression

### Checklist ###
  - [ ] test case
  - [ ] doc

### CODE REVIEW

**Comment by liuzhe-lz** [comment]
https://nni.readthedocs.io/en/stable/notes/contributing.html#writing-new-tutorials

File: dependencies/recommended.txt [hunk 1]
```diff
@@ -21,3 +21,4 @@ matplotlib
 git+https://github.com/microsoft/nn-Meter.git#egg=nn_meter
 sympy
 timm >= 0.5.4
+datasets == 2.10.1
```
Comment (matluster) [inline dependencies/recommended.txt:24]: Any particular reason to freeze the version?

Comment (Bonytu) [reply dependencies/recommended.txt:24]: good suggestion, there is no need to freeze the version. I will modify it.

File: dependencies/recommended.txt [hunk 2]
```diff
@@ -21,3 +21,4 @@ matplotlib
 git+https://github.com/microsoft/nn-Meter.git#egg=nn_meter
 sympy
 timm >= 0.5.4
+datasets
```
Comment (J-shang) [inline dependencies/recommended.txt:24]: why datasets is recommended, but transformers not?

File: docs/source/examples.rst [hunk 3]
```diff
@@ -82,3 +82,11 @@ More examples can be found in our :githublink:`GitHub repository <examples>`.
    :image: ../img/thumbnails/pruning-tutorial.svg
    :background: indigo
    :tags: Compression
+
+.. cardlinkitem::
+   :header: Quantize Bert on Task MNLI
+   :description: An end to end example for how to using NNI to quantize transformer and show the real speedup number
```
Comment (J-shang) [inline docs/source/examples.rst:88]: can we real speedup?😂

File: examples/tutorials/quantization_bert_glue.py [hunk 4]
```diff
@@ -0,0 +1,304 @@
+"""
+Quantize BERT on Task GLUE
+==========================
+
+Here we show an effective transformer simulated quantization process that NNI team has tried, and users can use NNI to discover better process
```
Comment (J-shang) [inline examples/tutorials/quantization_bert_glue.py:5]: `.`

File: examples/tutorials/quantization_bert_glue.py [hunk 5]
```diff
@@ -0,0 +1,304 @@
...
+
+we use the BERT model and the trainer pipeline in the Transformers to do some experiments.
```
Comment (J-shang) [inline examples/tutorials/quantization_bert_glue.py:7]: `We`

File: examples/tutorials/quantization_bert_glue.py [hunk 6]
```diff
@@ -0,0 +1,304 @@
...
+
+    return train_dataset, validation_datasets
+
+# %%
+# Create a trainer instance
+
+# .. note::
+
+#     Please set ``is_trace`` to ``False`` to fine-tune the BERT model and set ``is_trace`` to ``True``
+#     When you need to create a traced trainer for model quantization.
```
Comment (J-shang) [inline examples/tutorials/quantization_bert_glue.py:137]: `, when`

Comment (J-shang) [reply examples/tutorials/quantization_bert_glue.py:137]: and fine-tuning model can also use a traced trainer.

File: examples/tutorials/quantization_bert_glue.py [hunk 7]
```diff
@@ -0,0 +1,304 @@
...
+
+import nni
+from nni.contrib.compression.quantization import QATQuantizer, LsqQuantizer, PtqQuantizer
+from nni.contrib.compression.utils import TransformersEvaluator
+
+def fake_quantize():
+    config_list = [{
+        'op_types': ['Linear'],
+        'op_names_re': ['bert.encoder.layer.{}'.format(i) for i in range(12)],
+        'target_names': ['weight', '_output_'],
```
Comment (J-shang) [inline examples/tutorials/quantization_bert_glue.py:227]: why not quant input?

File: examples/tutorials/quantization_bert_glue.py [hunk 8]
```diff
@@ -0,0 +1,304 @@
...
+def evaluate():
+    model = build_finetuning_model(f'./output/bert_finetuned/{task_name}.bin', is_trace=False)
+    trainer = prepare_traced_trainer(model, is_trace=False)
+    metrics = trainer.evaluate()
+    print(f"Evaluate metrics={metrics}")
+
+
+if __name__ == "__main__":
+    fake_quantize()
+    evaluate()
```
Comment (J-shang) [inline examples/tutorials/quantization_bert_glue.py:263]: remove `if __name__ == "__main__":`, this is a ipython style script.

**Review by ultmaster — APPROVED** [review]

File: docs/source/examples.rst (hunk 3 above)
Comment (Bonytu) [reply docs/source/examples.rst:88]: no no no

**Review by J-shang — APPROVED** [review]
will review this doc after merge

**Pull request merged**